File: backend/app/schemas/video.py
import uuid
from datetime import datetime

from pydantic import BaseModel


class VideoGenerateRequest(BaseModel):
    project_id: uuid.UUID
    languages: list[str] = ["en"]
    aspect_ratio: str = "16:9"
    resolution: str = "1080p"
    ai_provider: str = "openai"
    tts_provider: str = "elevenlabs"


class VideoResponse(BaseModel):
    id: uuid.UUID
    project_id: uuid.UUID
    aspect_ratio: str
    resolution: str
    duration_sec: float | None
    thumbnail_url: str | None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class VideoVariantResponse(BaseModel):
    id: uuid.UUID
    video_id: uuid.UUID
    language: str
    file_url: str | None
    title: str | None
    description: str | None
    hashtags: list[str] | None
    status: str
    created_at: datetime

    model_config = {"from_attributes": True}


class AddVariantRequest(BaseModel):
    language: str
    tts_provider: str = "elevenlabs"
    tts_voice_id: str | None = None

File: frontend/src/types/video.ts
export interface Project {
  id: string
  title: string
  original_prompt: string
  analyzed_content: Record<string, unknown> | null
  template_id: string | null
  status: 'draft' | 'processing' | 'ready' | 'published' | 'failed'
  settings: Record<string, unknown>
  created_at: string
}

export interface Video {
  id: string
  project_id: string
  aspect_ratio: string
  resolution: string
  duration_sec: number | null
  thumbnail_url: string | null
  status: string
  created_at: string
}

export interface VideoVariant {
  id: string
  video_id: string
  language: string
  file_url: string | null
  title: string | null
  description: string | null
  hashtags: string[] | null
  status: string
  created_at: string
}

export interface Template {
  id: string
  name: string
  category: string | null
  description: string | null
  scene_structure: Record<string, unknown>
  default_settings: Record<string, unknown>
  is_system: boolean
  created_at: string
}

export interface VideoGenerateRequest {
  project_id: string
  languages: string[]
  aspect_ratio: string
  resolution: string
  ai_provider: string
  tts_provider: string
}
